libs/platform/src/lib/components/form/switch/switch/switch.component.ts

```diff
@@ -0,0 +1,129 @@
+import {
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    EventEmitter,
+    Input,
+    NgZone,
+    Optional,
+    Output,
+    Self,
+    ViewEncapsulation
+} from '@angular/core';
+import { NgControl, NgForm } from '@angular/forms';
+
+import { ContentDensity, FormFieldControl } from '../../form-control';
+import { BaseInput } from '../../base.input';
+import { SwitchConfig } from './switch.config';
+
+/** Switch change event instance */
+export class SwitchChangeEvent {
+    /** The source Switch of the event. */
+    source: SwitchComponent;
+    /** The new `payload` value of the switch. */
+    payload: boolean;
+}
+
+@Component({
+    selector: 'fdp-switch',
+    templateUrl: './switch.component.html',
+    styleUrls: ['./switch.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None,
+    providers: [{ provide: FormFieldControl, useExisting: SwitchComponent, multi: true }]
+})
+export class SwitchComponent extends BaseInput {
+    /** aria-label attribute of the inner input element. */
+    @Input()
+    ariaLabel = 'Switch input';
```

Comment:
I know it's detail, but could you add type there?

Reply:
In this you dont really need to. as you assign a string, this is default behavior for Lint Rules. We dont have general guidelines for this and this is some thing to be noted as well and decide on the direction. I would leave this as it is as this is also proper form.